### PR TITLE
Include program + type when available in fetchJsonParsedAccount

### LINKED ISSUE #1151: Include program + type when available in `fetchJsonParsedAccount`
When `jsonParsed` successfully decodes accounts for core programs, it includes a `program` and `type` field:

```json
{
    "value": {
      "data": {
        "parsed": {
          "info": {
            "decimals": 6,
            "freezeAuthority": "7dGbd2QZcCKcTndnHcTL8q7SMVXAkp688NTQYwrRCrar",
            "isInitialized": true,
            "mintAuthority": "BJE5MMbqXjVwjAF7oxwPYXnTXDyspzZyt4vwenNw5ruG",
            "supply": "11457520777246667"
          },
          "type": "mint" <--
        },
        "program": "spl-token", <--
        "space": 82
      },
      "executable": false,
      "lamports": 427263616843,
      "owner": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA",
      "rentEpoch": 18446744073709551615,
      "space": 82
    }
  }
}
```

But when we use `fetchJsonParsedAccount` we only include `parsed.info` as data

https://github.com/anza-xyz/kit/blob/84b2f81e76fa9a2675db6743e48941d4d2679ab9/packages/accounts/src/parse-account.ts#L102-L109

I think it'd be helpful to also include `program` and `type`, probably both as `string?` in the return type

Maybe we could return `MaybeAccount<TData & { program?: string, type?: string } , TAddress>`? 

Then callers would have access to these fields if they were returned by `jsonParsed`, which is useful for core programs.

### PR DESCRIPTION
**Pull request by rajgoesout**
#### Problem
When using [fetchJsonParsedAccount](https://github.com/anza-xyz/kit/blob/84b2f81e76fa9a2675db6743e48941d4d2679ab9/packages/accounts/src/fetch-account.ts#L89), we just get `parsed.info` as data; `program` and `type` are not included even if available.

#### Summary of Changes
Added `program`/`type` to jsonParsed account data (optional in types), updated fetch/parse return types, and adjusted tests (including a missing-info edge case) to reflect the new metadata.

Closes #1151

### CODE REVIEW

File: packages/accounts/src/parse-account.ts [hunk 1]
```diff
@@ -76,6 +76,7 @@ export function parseBase58RpcAccount<TAddress extends string = string>(
 }
 
 type JsonParsedRpcAccount = AccountInfoBase & { readonly data: JsonParsedDataResponse<unknown> };
+type JsonParsedAccountData<TData extends object> = TData & { program?: string; type?: string };
```
Comment (mcintyre94) [inline packages/accounts/src/parse-account.ts]: I think we should make this type a bit more strict, because you only have a type if there is a program. If we move this into its own optional field, then we could also remove the empty case. I think we could have:

```ts
type ParsedAccountMeta = { program: string, type?: string }
type  JsonParsedAccountData<TData extends object> = TData & { parsedAccountMeta?: ParsedAccountMeta };
```

**Comment by rajgoesout** [comment]
@mcintyre94 makes sense. BTW should we keep the `parsedAccountMeta` field outside the `data` field as well?

**Comment by mcintyre94** [comment]
@rajgoesout If we were doing this from scratch then I might think about using a new `Account` type and taking it outside of data, but I don't think it's worth that big a change here. We're good to add `parsedAccountMeta` to data.

**Comment by rajgoesout** [comment]
@mcintyre94 

CI was failing because adding `parsedAccountMeta` caused JSON‑parsed array data to be turned into objects like `{ "0": ..., "1": ... }`. That broke array‑shaped sysvars (recent blockhashes, slot hashes, stake history) against the toMatchObject assertions. (`sysvar-test.ts`)

Screenshot 2026-01-12 at 10 33 16 PM

lmk if this fix makes sense: https://github.com/anza-xyz/kit/pull/1152/commits/22b9748550125dd388a33eb0bd619ab8d0cac3db

File: packages/accounts/src/parse-account.ts [hunk 2]
```diff
@@ -94,17 +96,36 @@ type JsonParsedRpcAccount = AccountInfoBase & { readonly data: JsonParsedDataRes
...
     if (!rpcAccount) return Object.freeze({ address, exists: false });
-    const data = rpcAccount.data.parsed.info as TData;
+    const parsed = rpcAccount.data.parsed;
+    const info = parsed.info as TData | undefined;
+    const parsedAccountMeta =
+        rpcAccount.data.program || parsed.type
+            ? { program: rpcAccount.data.program, ...(parsed.type ? { type: parsed.type } : {}) }
+            : undefined;
+    let data: JsonParsedAccountData<TData>;
+    if (parsedAccountMeta) {
```
Comment (mcintyre94) [inline packages/accounts/src/parse-account.ts]: Hmm good spot, I hadn't thought about how sysvars use this. Could you add some more unit tests in `parse-account-test` to capture the different cases here?

This function is nice to test because we can control the `rpcAccount` input so we should be able to poke at each scenario here in isolation.

Comment (rajgoesout) [reply packages/accounts/src/parse-account.ts]: added
